server/expression.js

```diff
@@ -357,12 +373,15 @@ expression.readMatrix_ = function(fileName, inputGenes,
     });
   }
   geneNames.forEach(function(geneName) {
-    var geneIndex = allGeneNames[geneName];
-    if (geneName in allGeneNames) {
+    var geneIndex = allGeneNames[geneName.toLowerCase()].index;
+    if (geneName.toLowerCase() in allGeneNames) {
       var parts = lines[geneIndex].split(/[\t\s]+/);
       var tmpLine = [];
       conditions.forEach(function(conditionIndex) {
         var value = parseFloat(parts[conditionIndex]);
+        if (!value) {
```

Comment:
I am not sure. I found out the empty and undefined parts are from expressionMatrix.bin or rnaseq.bin. They are both binary files. I don't have original files and cannot check the archetype.